src/vs/workbench/contrib/bulkEdit/browser/bulkEditService.ts

```diff
@@ -88,6 +88,10 @@ class BulkEdit {
 		}
 	}
 
+	isApplied(): boolean {
+		return this._edits.length > 0;
+	}
```

Comment:
I'd arguee that this isn't needed. Can't you check the `edit.length` again, after calling the preview handler and bail if having none?

Reply:
I think you are right! Updated to use the edits directly in the `apply()`.

Reply:
I think you are right. Updated :)